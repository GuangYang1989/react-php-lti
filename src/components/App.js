import React from 'react';
import styled from 'styled-components';

import axios from 'axios';

const AppContainer = styled.div`
    border: 1px solid lightgrey;
    padding: 50px;
    height: 100px;
`;

const App = () => (
    <AppContainer>
        <h3>Hello React App!</h3>
        <button
            type="button"
            onClick={() => {
                axios({
                    method: 'get',
                    url: '../public/api/api.php',
                    params: {
                        action: 'hello',
                        jwt_token: $JWT_TOKEN,
                        data: {
                            name: $LTI.userID,
                        },
                    },
                })
                    .then(response => {
                        console.log(response);
                    })
                    .catch(error => {
                        console.log(error);
                    });

                axios({
                    method: 'get',
                    url: '../public/api/crud.php/posts', // test with posts table
                    params: {
                        jwt_token: $JWT_TOKEN,
                        filter: 'posts.title,eq,A post by Will',
                        transform: 1,
                    },
                })
                    .then(response => {
                        console.log(response);
                    })
                    .catch(error => {
                        console.log(error);
                    });

                console.log('wow');
            }}
        >
            Test API
        </button>
    </AppContainer>
);

export default App;
