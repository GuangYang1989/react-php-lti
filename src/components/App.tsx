import * as React from "react";
import styled from "styled-components";
import axios from "axios";
import { withRouter } from "react-router-dom";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

import Counter from "./Counter";

interface APIButtonComponentProps {
  onClick: { (): void };
}

const AppContainer = styled.div`
  border: 1px solid lightgrey;
  padding: 50px;
`;

const AppTitle = styled.div`
  font-weight: bold;
  font-size: 25px;
`;

const APIButtonMessage = styled.div``;

const APIButton = styled.button``;

const APIButtonComponent = ({ onClick }: APIButtonComponentProps) => (
  <APIButton type="button" onClick={onClick}>
    Test API
  </APIButton>
);

const App = () => (
  <AppContainer>
    <AppTitle>
      Hello LTI
      <FontAwesomeIcon icon={["fab", "react"]} />
      React App!
    </AppTitle>
    <p>
      Open <FontAwesomeIcon icon="cog" /> Developer Tools to see log outputs on
      state change
    </p>

    <APIButtonMessage>
      Ensure that you have created the sample database and tables to test the
      API button below, Use developer tools to see output.
    </APIButtonMessage>
    <APIButtonComponent
      onClick={() => {
        const a = { red: "green", blue: "orange" };
        const b = { ...a, blue: "red" };
        console.log(b);
        axios({
          method: "get",
          url: "../public/api/api.php",
          params: {
            action: "hello",
            jwt_token: $JWT_TOKEN,
            data: {
              name: $LTI.userID
            }
          }
        })
          .then(response => {
            console.log(response);
          })
          .catch(error => {
            console.log(error);
          });

        axios({
          method: "get",
          url: "../public/api/crud.php/posts", // test with posts table
          params: {
            jwt_token: $JWT_TOKEN,
            filter: "posts.title,eq,A post by Will",
            transform: 1
          }
        })
          .then(response => {
            console.log(response);
          })
          .catch(error => {
            console.log(error);
          });
      }}
    />
    <h4>
      <FontAwesomeIcon icon="store" /> Redux Example
    </h4>
    <Counter />
  </AppContainer>
);

export { APIButtonComponent };

export default withRouter(App);
