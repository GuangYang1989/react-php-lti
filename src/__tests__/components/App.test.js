import React from 'react';
import {
    shallow,
    // mount,
    // render
} from 'enzyme';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import 'jest-styled-components';
import App from '../../components/App';

it('adds 1 + 2 to equal 3', () => {
    expect(1 + 2).toEqual(3);
});

it('renders successfully', () => {
    const wrapper = shallow(<App.WrappedComponent />);

    expect(wrapper).toMatchSnapshot();
});
it('has title', () => {
    const wrapper = shallow(<App.WrappedComponent />);

    expect(wrapper.find('AppTitle')).toHaveLength(1);
});

describe('App Component Title', () => {
    const wrapper = shallow(<App.WrappedComponent />);
    const title = wrapper.find('AppTitle');

    it('title has 2 icons', () => {
        expect(title.find(FontAwesomeIcon)).toHaveLength(2);
    });

    it('has correct text', () => {
        expect(title.childAt(0).text()).toBe('Hello LTI ');
        expect(title.childAt(2).text()).toBe('React App!');
    });

    it('has correct icons', () => {
        const icons = title.findWhere(n => n.type() === FontAwesomeIcon);
        expect(icons.at(0).props().icon[1]).toEqual('react');
        expect(icons.at(1).props().icon[1]).toEqual('cog');
    });
});

describe('Test Api Button', () => {
    const wrapper = shallow(<App.WrappedComponent />);
    const component = wrapper.find('TestAPIButton');

    it('Has correct styling', () => {
        expect(component).toHaveStyleRule('color', 'white');
        expect(component).toHaveStyleRule('background-color', '#4caf50');
        expect(component).toHaveStyleRule('border', 'none');
        expect(component).toHaveStyleRule('padding', '15px 32px');
        expect(component).toHaveStyleRule('text-align', 'center');
        expect(component).toHaveStyleRule('text-decoration', 'none');
        expect(component).toHaveStyleRule('display', 'inline-block');
        expect(component).toHaveStyleRule('font-size', '16px');
    });

    it('Has Correct Label', () => {
        expect(component.children().text()).toEqual('Test API');
    });
});
