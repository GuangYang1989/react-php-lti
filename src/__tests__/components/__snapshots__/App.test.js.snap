// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches snapshot 1`] = `
<AppContainer>
  <AppTitle>
    Hello LTI
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fab",
          "react",
        ]
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    React App!
  </AppTitle>
  <p>
    Open 
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="cog"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
     Developer Tools to see log outputs on state change
  </p>
  <APIButtonMessage>
    Ensure that you have created the sample database and tables to test the API button below, Use developer tools to see output.
  </APIButtonMessage>
  <APIButtonComponent
    onClick={[Function]}
  />
  <h4>
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="store"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
     Redux Example
  </h4>
  <h5>
    PHP MESSAGE: 
  </h5>
  <p>
    DB POST: 
  </p>
  <h5>
    COUNTER EXAMPLE
  </h5>
  <p>
    APP COMPONENT: 
    0
  </p>
  COUNTER COMPONENT: 
  <Connect(CounterComponent) />
</AppContainer>
`;
